File: python_july/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_max = global_max = nums[0]
        n = len(nums)

        for i in range(1, n):
            current_max = max(nums[i], current_max+nums[i])
            if current_max > global_max:
                global_max = current_max
        
        return global_max

File: java_july/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = nums[0], curr = 0;
        int n = nums.length;
        for (int num : nums) {
            curr += num;
            max = max > curr ? max : curr;
            if (curr < 0) {
                curr = 0;
            }
        }
        return max;
    }
}